src/steps/assets.ts

```diff
@@ -22,25 +22,29 @@ export async function fetchAssets({
   await apiClient.iterateAssets(async (asset) => {
     const webLink = asset.links.find((link) => link.rel === 'self')?.href;
 
-    const assetEntity = createIntegrationEntity({
-      entityData: {
-        source: {},
-        assign: {
-          _key: getAssetKey(asset.id),
-          _type: entities.ASSET._type,
-          _class: entities.ASSET._class,
-          id: `${asset.id}`,
-          name: asset.hostName,
-          category: 'server',
-          make: 'unknown',
-          model: 'unknown',
-          serial: 'unknown',
-          webLink,
+    if (asset.hostName != undefined) {
+      const assetEntity = createIntegrationEntity({
+        entityData: {
+          source: asset,
+          assign: {
+            _key: getAssetKey(asset.id),
+            _type: entities.ASSET._type,
+            _class: entities.ASSET._class,
+            id: `${asset.id}`,
+            name: asset.hostName,
+            osName: asset.os, // in our case r7 does not always correctly identify the OS
+            ipAddress: asset.ip,
+            category: 'server',
+            make: 'unknown',
+            model: 'unknown',
+            serial: 'unknown',
```

Comment:
```suggestion
            make: null,
            model: null,
            serial: null,
```
These are required properties for the `Device` class, but if we aren't able to satisfy them, we should `null` them. If you run the integration with `yarn start` and it complains about entities failing validation, we should change the `start` command in `package.json` from:
```json
"start": "j1-integration collect"
``` 
to:
```json
"start":"j1-integration collect --disable-schema-validation"
```

**Note**: The integration shouldn't report issues when `make`, `model`, or `serial` are set to `null`. I believe this is actually a bug in the `data-model` or `sdk` that I should create an issue for. It is annoying to see that happen. 🤔

Reply:
If those three properties are always `null` I guess we might as well just remove them entirely, right? As mentioned above, I changed the `start` command like you suggested.

Reply:
We should try to fill in required properties when we can. Here we don't have the data so we should explicitly mark them `null`. This communicates to the `data-mode` and to others that we tried adding this to match the `data-model`, but it just wasn't possible.

When we mark it as `null` the data-model will ignore that is missing and allow us to use other nice features like the schema validation and other testing functions. Without it these nice features won't work properly.

Reply:
Also I didn't realize that this was the code already in place. This integration is out-of-date, so making these changes will help a lot towards getting it back in line.